Sources/Networking/Core/DownloadAPIManager.swift

```diff
@@ -156,47 +157,45 @@ private extension DownloadAPIManager {
     /// Creates a record in the `downloadStateDict` for each task and observes their states.
     /// Every `downloadStateDict` update triggers an event to the `downloadStateDictSubject`
     /// which then leads to a task state update from `progressStream`.
-    func updateTasks() {
-        Task {
-            for task in await allTasks where await downloadStateDict.getValue(for: task) == nil {
-                /// In case there is no DownloadState for a given task in the dictionary, we need to create one.
-                await downloadStateDict.set(value: .init(task: task), for: task)
-                
-                /// We need to observe URLSessionTask.State via KVO individually for each task, because there is no delegate callback for the state change.
-                let cancellable = task
-                    .publisher(for: \.state)
-                    .sink { [weak self] state in
-                        guard let self else {
-                            return
-                        }
-                        
-                        Task {
-                            await self.downloadStateDict.update(task: task, for: \.taskState, with: state)
-                            self.downloadStateDictSubject.send(await self.downloadStateDict.getValues())
-                            
-                            if state == .completed {
-                                await self.taskStateCancellables.set(value: nil, for: task)
-                            }
+    func updateTasks() async {
+        for task in await allTasks where await downloadStateDict.getValue(for: task) == nil {
+            /// In case there is no DownloadState for a given task in the dictionary, we need to create one.
+            await downloadStateDict.set(value: .init(task: task), for: task)
+
+            /// We need to observe URLSessionTask.State via KVO individually for each task, because there is no delegate callback for the state change.
+            let cancellable = task
+                .publisher(for: \.state)
+                .sink { [weak self] state in
```

Comment:
It's shorter? 😄 What is the benefit of doing it the other way?

Reply:
Visibility of what is being captured 😄, smaller memory footprint (although this really doesn't make a difference 😄) . But we would most likely have to capture everything weakly and there would be a long unwrap. So all good with `self` 👍 . 

I had it there mainly, because `Task` was screaming at me that I'm capturing non-sendable self 😄 and by capturing only Sendable properties, it was happy.

Reply:
I see 😄